frame/staking/src/lib.rs

```diff
@@ -1755,6 +1781,13 @@ decl_module! {
 		/// This can be called from any origin.
 		///
 		/// - `stash`: The stash account to reap. Its balance must be zero.
+		///
+		/// # <weight>
+		/// - `O(1)`
+		/// - One slash kill
+		/// - Two lock removes
+		/// # </weight>
+		#[weight = SimpleDispatchInfo::FixedOperational(10_000)]
```

Comment:
I have question that why original didn't require weight? Will unsigned extrinsics successful?

Need someone else to review this.

Reply:
I'm not sure about this, so applied the lowest weight in darwinia weight standard, but this method seems not completed yet.

```rust
#[weight = SimpleDispatchInfo::FixedOperational(10_000)]
fn reap_stash(_origin, stash: T::AccountId) {
    Self::ensure_storage_upgraded();
    ensure!(T::RingCurrency::total_balance(&stash).is_zero(), <Error<T>>::FundedTarget);
    ensure!(T::KtonCurrency::total_balance(&stash).is_zero(), <Error<T>>::FundedTarget);

    Self::kill_stash(&stash)?;
    T::RingCurrency::remove_lock(STAKING_ID, &stash);
    T::KtonCurrency::remove_lock(STAKING_ID, &stash);
 }
```

However, the `origin` is `_origin` and doesn't have a `ensure!`, so the system might not know how to spend weights, it might be safer to keep no weight costs as the original implementation.

Reply:
How will the system calculate the fees if we don't set the weight tags?

Reply:
The default is ```SimpleDispatchInfo::FixedNormal(10_000)```

Refer https://zhuanlan.zhihu.com/p/108194544 and the code: 
https://github.com/paritytech/substrate/blob/master/frame/support/src/weights.rs#L233

> 在有限的区块生成时间和链上状态的限制下，权重被用来定义交易产生的计算复杂度即所消耗的计算资源，以及占据的链上状态。system模块定义了区块的总权重（MaximumBlockWeight）。为了保证在网络繁忙的情况下，依然能够实现对区块链应用有效合理的管理，Substrate引入了两种不同级别的交易类型，既 Normal 和 Operational。Normal类型的交易是由网络中的普通用户提交，Operational类型的交易是由网络中的管理员或者管理委员会共同触发。区块资源如长度和总权重按照一定比例在这两种类型的交易中进行分配，这一比例称为可用区块比（AvailableBlockRatio）。Kusama网络的设置为：
> 
> 区块的总权重：1,000,000,000
> 可用区块比：75%，即Normal交易最多只占用75%的区块资源，Operational类型的交易则可以占用100%的区块资源，新的交易如果导致对应资源使用率超过阈值后，会被拒绝。

FixNormal and Operational is for different purpose, so we should not use Operational here.

10_000 or default (not set) is both OK.

Reply:
Get it, I'll apply these explainations to the-darwinia-book's weight chapter.